domain/secretbackend/state/state.go

```diff
@@ -0,0 +1,562 @@
+			err = tx.Query(ctx, configStmt,
+				SecretBackendConfig{
+					BackendUUID: backend.ID,
+					Name:        k,
+					Content:     v.(string),
+				},
+			).Run()
+			if err != nil {
+				return handleSecretBackendConfigError(err, backend.ID)
+			}
+		}
+		return nil
+	})
+	return domain.CoerceError(err)
+}
+
+// DeleteSecretBackend deletes the secret backend for the given backend ID.
+func (s *State) DeleteSecretBackend(ctx context.Context, backendID string, force bool) error {
+	db, err := s.DB()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	// TODO: check if the backend is in use
+	// if !force {
+	// }
+	cfgStmt, err := sqlair.Prepare(`
+DELETE FROM secret_backend_config WHERE backend_uuid = $M.uuid`, sqlair.M{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	rotationStmt, err := sqlair.Prepare(`
+DELETE FROM secret_backend_rotation WHERE backend_uuid = $M.uuid`, sqlair.M{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	backendStmt, err := sqlair.Prepare(`
+DELETE FROM secret_backend WHERE uuid = $M.uuid`, sqlair.M{})
+	if err != nil {
+		return errors.Trace(err)
+	}
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		arg := sqlair.M{"uuid": backendID}
+		if err := tx.Query(ctx, cfgStmt, arg).Run(); err != nil {
+			return fmt.Errorf("deleting secret backend config for %q: %w", backendID, err)
+		}
+		if err := tx.Query(ctx, rotationStmt, arg).Run(); err != nil {
+			return fmt.Errorf("deleting secret backend rotation for %q: %w", backendID, err)
+		}
+		if err = tx.Query(ctx, backendStmt, arg).Run(); err != nil {
+			return fmt.Errorf("deleting secret backend for %q: %w", backendID, err)
+		}
+		return nil
+	})
+	return domain.CoerceError(err)
+}
+
+// ListSecretBackends returns a list of all secret backends.
+func (s *State) ListSecretBackends(ctx context.Context) ([]*secretbackend.SecretBackendInfo, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	stmt, err := sqlair.Prepare(`
+SELECT 
+    b.uuid                  AS &SecretBackendRow.uuid,
+    b.name                  AS &SecretBackendRow.name,
+    b.backend_type          AS &SecretBackendRow.backend_type,
+    b.token_rotate_interval AS &SecretBackendRow.token_rotate_interval,
+    c.name                  AS &SecretBackendRow.config_name,
+    c.content               AS &SecretBackendRow.config_content
+FROM secret_backend b
+    LEFT JOIN secret_backend_config c ON b.uuid = c.backend_uuid
+ORDER BY b.name`, SecretBackendRow{})
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var rows SecretBackendRows
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt).GetAll(&rows)
+		if err != nil {
+			return fmt.Errorf("querying secret backends: %w", err)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("cannot list secret backends: %w", err)
+	}
+	var result []*secretbackend.SecretBackendInfo
+	for _, backend := range rows.ToSecretBackendInfo() {
```

Comment:
`rows.ToSecretBackendInfo()` returns a slice of infos. I don't get why we need a for loop here.